README.md

```diff
@@ -37,4 +43,87 @@ Build targets are: net40, net45, netstandard1.4, netstandard2.1, and netcoreapp2
 
 Visual Studio 2019 is required for executing the unit tests. Requirements:
 - NUnit test adaptor (VS Extension)
-- Be sure to set the default processor architecture for tests to x64 (or giant allocation test will fail)
+- Be sure to set the default processor architecture for tests to x64 (or the giant allocation test will fail)
+
+## How It Works
+
+RecyclableMemoryStream improves GC performance by ensuring that the larger buffers used for the streams are put into the gen 2 heap and stay there forever. This should cause full collections to happen less frequently. If you pick buffer sizes above 85K then you will ensure these are placed on the large object heap, which is touched even less frequently by the garbage collector.
+
+The MemoryStreamManager class maintains two separate pools of objects:
+
+1. **Small Pool** - Holds small buffers (of configurable size). Used by default for all normal read/write operations. Multiple small buffers are chained together in the RecyclableMemoryStream class and abstracted into a single stream.
+2. ** Large Pool** - Holds large buffers, which are only used when you must have a single, contiguous buffer, such as a call to GetBuffer(). 
+
```

Comment:
Perhaps worth mentioning at this point: buffers above the max large pool size (i.e. the ones that fire the event MemoryStreamNonPooledLargeBufferCreated) will simply be allocated as if it were a normal MemoryStream, and the buffer will be eligible for garbage collection when it is disposed / goes out of scope.

One of the nice implications of this design is that if you have to allocate a very large stream, it doesn't steal buffers from the bounded pools. (I helped someone with this problem once, where they were concerned that the occasional huge RMS, like 100+MB, would starve the RMS manager).